contracts/token/ERC20/extensions/ERC20Capped.sol

```diff
@@ -32,7 +44,11 @@ abstract contract ERC20Capped is ERC20 {
      */
     function _update(address from, address to, uint256 amount) internal virtual override {
         if (from == address(0)) {
-            require(totalSupply() + amount <= cap(), "ERC20Capped: cap exceeded");
+            uint256 maxSupply = cap();
+            uint256 increasedSupply = totalSupply() + amount;
+            if (increasedSupply > maxSupply) {
+                revert ERC20ExceededCap(increasedSupply, maxSupply);
+            }
```

Comment:
In ERC20Votes we do the check after the update. Here we do it before. I think both check should be done the same way.

Reply:
I updated following the suggestion, but I'm leaving it open just to double-check with @frangio if there's no unexpected consequence of changing the order.

I tried to think of an edge case with the inheritance order in which the check would behave differently, but I didn't think of anything that could be reasonably bad, do you guys see something?

Reply:
My point is that the same order should be used in both cases. If we believe the right way is to check before the update, then ok ... but lets do it the same way in both locations.

Reply:
Nice, I agree with having both in the same order. Solving since that's the current state.